### ISSUE TITLE
get_child(0) throws error in console

### ISSUE THREAD

**Issue by burstina** [godotengine/godot#52682]
### Godot version

3.3.3 stable

### System information

Windows 10

### Issue description

get_child(0) method throws this message in console window:

`ERROR: get_child: Index p_index = 0 is out of bounds (data.children.size() = 0).`

Techincally I could override this message with

```
var aa
if get_child_count() > 0:
   aa=get_child(0) 
else:
   aa=null
```

but i am concerned about performance degradation as nodes grow in number.

### Steps to reproduce

run mrp

### Minimal reproduction project

[child_0.zip](https://github.com/godotengine/godot/files/7164729/child_0.zip)
[](url)

**Comment by pycbouh** [comment]
Calling methods with invalid parameters does indeed result in errors and warnings in the output log. You should check the number of children, or their existence, before trying to access any.

**Comment by beseeley** [comment]
Similarly get_child_count() on an empty container still returns 1. This combined with above tells me that all objects have a child count of at least 1 even when empty? Possibly itself?

**Comment by beseeley** [comment]
I am using a Node as a container for instanced scenes, coins in this case. I print its child count at beginning and it is 0. Later I add objects to it, and when all objects are removed it still returns a child count of 1. I've paused the game at that point to verify it is in fact empty. It is entirely possible I got a bug somewhere else in my code.

I'm working on making a minimum to reproduce to test this out.
